app/src/androidTest/java/org/mozilla/focus/activity/GrantStoragePermissionTest.java

```diff
@@ -0,0 +1,172 @@
+                        .setBody(AndroidTestUtils.readTestAsset(IMAGE_FILE_NAME_DOWNLOADED)));
+                webServer.enqueue(new MockResponse()
+                        .setBody(AndroidTestUtils.readTestAsset(IMAGE_FILE_NAME_DOWNLOADED)));
+
+                webServer.start();
+            } catch (IOException e) {
+                throw new AssertionError("Could not start web server", e);
+            }
+        }
+
+        @Override
+        protected void afterActivityFinished() {
+            super.afterActivityFinished();
+            try {
+                webServer.close();
+                webServer.shutdown();
+            } catch (IOException e) {
+                throw new AssertionError("Could not stop web server", e);
+            }
+        }
+    };
+
+    @Before
+    public void setUp() {
+        context = InstrumentationRegistry.getInstrumentation().getTargetContext();
+        AndroidTestUtils.beforeTest();
+
+    }
+
+    @After
+    public void tearDown() {
+        if (sessionLoadedIdlingResource != null) {
+            IdlingRegistry.getInstance().unregister(sessionLoadedIdlingResource);
+        }
+        activityRule.getActivity().finishAndRemoveTask();
+    }
+
+    @Test
+    public void downloadImageAndDelete_deleteSuccessfully() {
+
+        // Start main activity
+        activityRule.launchActivity(new Intent());
+        sessionLoadedIdlingResource = new SessionLoadedIdlingResource(activityRule.getActivity());
+
+        // Click home search field
+        onView(withId(R.id.home_fragment_fake_input)).perform(click());
+
+        // Enter URL and load the page
+        onView(withId(R.id.url_edit)).perform(replaceText(webServer.url(TEST_PATH).toString()), pressImeActionButton());
+
+        // Waiting for the page is loaded
+        IdlingRegistry.getInstance().register(sessionLoadedIdlingResource);
+
+        // Since test web page has a fullscreen image, we simulate the long click event at the center of image
+        // Long click on image with popup context menu
+        final DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();
+        final int displayWidth = displayMetrics.widthPixels;
+        final int displayHeight = displayMetrics.heightPixels;
+        onView(withId(R.id.main_content)).check(matches(isDisplayed())).perform(AndroidTestUtils.clickXY(displayWidth / 2, displayHeight / 2, Tap.LONG));
+
+        // Click save image button
+        onView(allOf(withText(R.string.contextmenu_image_save), isDisplayed())).perform(click());
+
+        // Open menu
+        // Since right now snackbar will overlap with menu bar and we don't want to wait until snackbar is dismissed,
+        // we cannot call onView(withId(R.id.btn_menu) here so call showMenu in MainActivity instead.
+        AndroidTestUtils.showHomeMenu(activityRule);
+
+        // Open download panel
+        onView(withId(R.id.menu_download)).check(matches(isDisplayed())).perform(click());
```

Comment:
If the test download file is huge, will the test fail here?
Will it be better to use idling resource here?